pkg/controller/supportbundlecollection/controller.go

```diff
@@ -667,13 +886,77 @@ func conditionExistsIgnoreLastTransitionTime(conditions []v1alpha1.SupportBundle
 	return false
 }
 
-func appendConditions(oldConditions, updatedConditions []v1alpha1.SupportBundleCollectionCondition) []v1alpha1.SupportBundleCollectionCondition {
-	newConditions := oldConditions
-	for _, c := range updatedConditions {
-		if conditionExistsIgnoreLastTransitionTime(newConditions, c) {
+func appendConditions(oldConditions, newConditions []v1alpha1.SupportBundleCollectionCondition) []v1alpha1.SupportBundleCollectionCondition {
+	finalConditions := make([]v1alpha1.SupportBundleCollectionCondition, 0)
+	newConditionMap := make(map[v1alpha1.SupportBundleCollectionConditionType]v1alpha1.SupportBundleCollectionCondition)
+	addedConditions := sets.NewString()
+	for _, condition := range newConditions {
+		newConditionMap[condition.Type] = condition
+	}
+	for _, oldCondition := range oldConditions {
+		newCondition, exists := newConditionMap[oldCondition.Type]
+		if !exists {
+			finalConditions = append(finalConditions, oldCondition)
 			continue
 		}
-		newConditions = append(newConditions, c)
+		// Use the original Condition if the only change is about lastTransition time
+		if conditionEqualsIgnoreLastTransitionTime(newCondition, oldCondition) {
+			finalConditions = append(finalConditions, oldCondition)
+		} else {
+			// Use the latest Condition.
+			finalConditions = append(finalConditions, newCondition)
+		}
+		addedConditions.Insert(string(newCondition.Type))
+	}
+	for key, newCondition := range newConditionMap {
+		if !addedConditions.Has(string(key)) {
+			finalConditions = append(finalConditions, newCondition)
+		}
+	}
+	return finalConditions
+}
+
+func getNodeKey(status *controlplane.SupportBundleCollectionNodeStatus) string {
+	// TODO: use NodeNamespace/NodeName for ExternalNode after the Namespace is added in the connection key between
+	// antrea-agent and antrea-controller.
+	return status.NodeName
+}
+
+// supportBundleCollectionStatusEqual compares two SupportBundleCollectionStatus objects. It disregards
+// the LastTransitionTime field in the status Conditions.
+func supportBundleCollectionStatusEqual(oldStatus, newStatus v1alpha1.SupportBundleCollectionStatus) bool {
+	semanticIgnoreLastTransitionTime := conversion.EqualitiesOrDie(
+		conditionSliceEqualsIgnoreLastTransitionTime,
+	)
```

Comment:
this should be declared as a global variable to avoid repeated execution as it's not specific to inputs.

Reply:
Updated.